Skip inlining Custom Task

vdemeester opened the pull request:
<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

# Changes

If a Custom Task is used in `PipelineRun` or `Pipeline`, the resolver
won't try to inline it. We detect Custom Task from their `apiVersion`
being not empty and not starting with `tekton.dev/`.

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>


# Submitter Checklist

- [x] ♽  Run `make test lint` before submitting a PR (ie: with [pre-commit](https://pipelinesascode.com/dev/tools), no need to waste CPU cycle on CI
- [x] 📖 If you are adding a user facing feature or make a change of the behavior, please verify that you have documented it
- [x] 🧪 100% coverage is not a target but most of the time we would rather have a unit test if you make a code change.
- [x] 🎁 If that's something that is possible to do please ensure to check if we can add a e2e test.
- [x] 🔎 If there is a flakiness in the CI tests then don't *necessary* ignore it, better get the flakyness fixed before merging or if that's not possible there is a good reason to bypass it. (token rate limitation may be a good reason to skip).

chmouel commented on docs/content/docs/guide/resolver.md:
```diff
@@ -24,9 +24,11 @@ annotations.
 [`Bundle`](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#tekton-bundles)
-(`Task` or `Pipeline`) the resolver will just use them as is and would not try
+ (`Task` or `Pipeline`)
+or a [Custom Task](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#using-custom-tasks)
```
why no tekton.dev links :-p

vdemeester replied:
to be constistent 😛 👼🏼

chmouel commented on docs/content/docs/guide/resolver.md:
```diff
@@ -24,9 +24,11 @@ annotations.
 [`Bundle`](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#tekton-bundles)
-(`Task` or `Pipeline`) the resolver will just use them as is and would not try
+ (`Task` or `Pipeline`)
+or a [Custom Task](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#using-custom-tasks)
```
```
The resolver will skip resolving if he sees these type of tasks:

* a reference to a [`ClusterTask`](https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md#task-vs-clustertask)
* a `Task` or `Pipeline` [`Bundle`](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#tekton-bundles)
* or a [Custom Task](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#using-custom-tasks) with an apiVersion that doesn't have a `tekton.dev/` prefix.

It just use them "as is" and will not try to do anything with it.
```

better phrasing

chmouel commented on pkg/resolve/resolve.go:
```diff
@@ -80,10 +80,15 @@ func skippingTask(taskName string, skippedTasks []string) bool {
 }
 
+func isTektonAPIVersion(apiVersion string) bool {
+	return strings.HasPrefix(apiVersion, "tekton.dev/") || apiVersion == ""
```
👍🏻

chmouel replied:
I think tekton.dev can be replaced by pipeline.GroupName or something

chmouel commented on pkg/resolve/resolve_test.go:
```diff
@@ -34,6 +34,7 @@ func setup() {
 
 // Not sure how to get testParams fixtures working
 func readTDfile(t *testing.T, testname string, generateName bool, remoteTasking bool) (*tektonv1beta1.PipelineRun, *zapobserver.ObservedLogs, error) {
+	t.Helper()
```
there is a bunch of them everywhere we didn't set as helper, we should update it.

<https://github.com/openshift-pipelines/pipelines-as-code/issues/657>

vdemeester replied:
yeah.. I shouldn't have commited this one here, and instead create a PR just for it but 😐

chmouel replied:
no worries...

vdemeester committed:
Skip inlining Custom Task

If a Custom Task is used in `PipelineRun` or `Pipeline`, the resolver
won't try to inline it. We detect Custom Task from their `apiVersion`
being not empty and not starting with `tekton.dev/`.

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>
